tactics/class_tactics.ml

```diff
@@ -930,8 +930,16 @@ module Search = struct
        let _, pv = Proofview.init evm [] in
        let pv = Proofview.unshelve goals pv in
        try
+         (* Instance may try to call this before a proof is set up!
+            Thus, give_me_the_proof will fail. Beware! *)
+         let name, poly = try
```

Comment:
Do you have an example execution which calls this without an open proof? I'd like to get an idea, it seems tricky.

Reply:
Can't tactics be executed by the pretyper (via TCs) without any open proof? Is this what is revealed here?

Reply:
Uhuh you are entering the Temple of Doom be wary the corridor is full of booby-traps.

This can be called by pretyping in many cases, such as to resolved typeclasses. IIRC, the standard library is enough to trigger it, in `RelationClasses`, just remove the check and see.

I have a tree with what could be a deeper fix, see https://github.com/ejgallego/coq/tree/make_tc_hook_tactic could be interesting to move that tree forward.

Reply:
> Can't tactics be executed by the pretyper (via TCs) without any open proof? Is this what is revealed here?

Yes, I saw your comment after I was writing mine. That's the thing. I conjecture the situation could be improved by the modification of the hooks done above. After all `engine` is below pretyping.

In the long term the hook should be removed and the pretyper should take maybe an additional parameter, but we need a lot of work on the upper layers to make this relizable IMO.

Reply:
If the pretyper was in the monad (or took name and poly as parameters) then the information would be available indeed.